### PR TITLE
Create binding to update a Foreman instance

### PR DESCRIPTION
**Pull request by maria**
## Need

``` gherkin
As a developer
I want to communicate with the Foreman instance via the Git repository
So that I can update the instance with the new Foreman configs
```
## Deliverables
- A manager which will handle the update of a Foreman instance with a given configuration
### Abstract

Using the ConfigurationManager load a list of components, and make API calls to Foreman to update/create resources based on the loaded components.
#### Prerequisites
- [Can Hammer be reused?](https://github.com/theforeman/hammer-cli-foreman)
- a class to upload configuration to Foreman through API endpoints (the same as above but with POST/PUT/PATCH) - [example](https://github.com/opscode/chef/blob/master/lib/chef/knife/upload.rb)
#### TODO
- [ ] Create tests for `push`
- [x] Create command method `push` which will make API calls to a Foreman instance 
#### Files
- lib/talk_commands.rb
- bin/foregit-talk

### CODE REVIEW

**Comment by maria** [comment]
started working on this

**Comment by maria** [comment]
TODOs:
- add tests
- split Foreman API action type based on the file status: 
  - changes -> update
  - new -> create
  - deleted -> delete

**Comment by maria** [comment]
Ivan, can you take a look please? It worked to update the resources. I still have to split the call action based on the changes, as stated above. The problem is that the git gem is quite limited, a bit hard to use, and I have to play with it a little longer.

**Comment by iNecas** [comment]
Hi, I hope to get into this today evening or tomorrow

File: config/settings.yaml.example [hunk 1]
```diff
@@ -8,3 +8,8 @@ api_user: 'admin'
 resources: :architectures
 # Git repository path
 repo_path: '/home/user/foregit/'
```
Comment (iNecas) [inline config/settings.yaml.example:10]: maybe using the relative path to the current foregit repo might be a bit more friendlier to the using directly from source, what about that being `./data` and having it in `.gitignore`?

Comment (iNecas) [inline config/settings.yaml.example:10]: although, let's postpone this change till the hammer plugin time

**Pull request merged**